### PR TITLE
Correct flags name and order of CBaseModelInfoSAInterface

### PR DESCRIPTION
**Pull request by thisdp**
Corrent some flags order and name of CBaseModelInfoSAInterface.

### CODE REVIEW

File: Client/game_sa/CModelInfoSA.cpp [hunk 1]
```diff
@@ -1292,13 +1292,16 @@ void CModelInfoSA::SetColModel(CColModel* pColModel)
         // If no collision model has been set before, store the original in case we want to restore it
         if (!m_pOriginalColModelInterface)
             m_pOriginalColModelInterface = m_pInterface->pColModel;
+            m_pInterface->bAlphaTransparency = false;
```
Comment (Lpsd) [inline Client/game_sa/CModelInfoSA.cpp]: This statement is not inside the condition above. I assume you want it to be based on the indentation?

A bracketless condition will only execute the first statement directly after it. If you have multiple statements, you need brackets.

```suggestion
        if (!m_pOriginalColModelInterface)
        {
            m_pOriginalColModelInterface = m_pInterface->pColModel;
            m_pInterface->bAlphaTransparency = false;
        }
```

File: Client/game_sa/CModelInfoSA.cpp [hunk 2]
```diff
@@ -1475,7 +1480,7 @@ void CModelInfoSA::MakeObjectModel(ushort usBaseID)
     MemCpyFast(m_pInterface, pBaseObjectInfo, sizeof(CBaseModelInfoSAInterface));
     m_pInterface->usNumberOfRefs = 0;
     m_pInterface->pRwObject = nullptr;
-    m_pInterface->usUnknown = 65535;
+    m_pInterface->uObjectInfoIndex = 65535;
```
Comment (Lpsd) [inline Client/game_sa/CModelInfoSA.cpp:1480]: Is this necessary to fixing the collision bug?

File: Client/game_sa/CModelInfoSA.cpp [hunk 3]
```diff
@@ -1493,7 +1498,7 @@ void CModelInfoSA::MakeVehicleAutomobile(ushort usBaseID)
     m_pInterface->usNumberOfRefs = 0;
     m_pInterface->pRwObject = nullptr;
     m_pInterface->pVisualInfo = nullptr;
-    m_pInterface->usUnknown = 65535;
+    m_pInterface->uObjectInfoIndex = 65535;
```
Comment (Lpsd) [inline Client/game_sa/CModelInfoSA.cpp:1498]: As above

File: Client/game_sa/CModelInfoSA.h [hunk 4]
```diff
@@ -151,34 +151,53 @@ class CBaseModelInfoSAInterface
...
+            unsigned char bHasComplexHierarchy : 1;             // CClumpModelInfo::SetHasComplexHierarchy(unsigned int)
+            unsigned char bUnknown1 : 1;
+            unsigned char bSwaysInWind : 1;                     // Confirmed: Sways in wind
+            unsigned char bCollisionWasStreamedWithModel : 1;   // CClumpModelInfo::SetCollisionWasStreamedWithModel(unsigned int)
+            unsigned char bOwnsCollisionModel : 1;
+            unsigned char bUnknown2 : 1;
+            unsigned char bTagDisabled : 1;
+        };
+    };
+
```
Comment (Lpsd) [inline Client/game_sa/CModelInfoSA.h:196]: Again, is any of this inside the scope of this PR? This should be submitted separately if it's not essential to fixing the collision bug.

File: Client/game_sa/CModelInfoSA.cpp (hunk 1 above)
Comment (Lpsd) [reply Client/game_sa/CModelInfoSA.cpp]: Also just noticed, is this relevant to fixing the bug either?

File: Client/game_sa/CModelInfoSA.cpp (hunk 3 above)
Comment (thisdp) [reply Client/game_sa/CModelInfoSA.cpp:1498]: correcting flag name

Comment (thisdp) [reply Client/game_sa/CModelInfoSA.cpp:1498]: > As above

The whole flags order of CBaseModelInfoSAInterface is at a mess 😂

Comment (Lpsd) [reply Client/game_sa/CModelInfoSA.cpp:1498]: but is this necessary to fix the collision glitch? if not, it needs to be in a separate PR

**Comment by Lpsd** [comment]
Submit a separate PR for the flags - just keep the code here necessary to fixing the collision glitch (or vice-versa)

**Comment by thisdp** [comment]
> Submit a separate PR for the flags - just keep the code here necessary to fixing the collision glitch (or vice-versa)

we need merge this first, and then the collision glitch fixing

**Pull request closed without merging**